### PR TITLE
[PATCH v5] README: improve api header README file

### PR DESCRIPTION
**Pull request by MatiasElo**
Add ODP API principles chapter and improve formatting of include/README file. The file is also referenced from application-api-guide and platform- api-guide for increased visibility.

### CODE REVIEW

File: doc/application-api-guide/api_guide_lines.dox [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 /**
 
-@page api_guide_lines  API Guide Lines
+@page api_guide_lines  API Developer Guide Lines
```
Comment (JereLeppanen) [inline doc/application-api-guide/api_guide_lines.dox:10]: I don't know if this is relevant to this PR, but "guidelines" is one word. There's at least one other occurrence (in application-api-guide).

File: include/README [hunk 2]
```diff
@@ -1,63 +1,104 @@
...
+not otherwise documented, results are undefined if an application acts against
+the specification. For example, if an application passes bad parameters to an
+ODP API, one implementation may report an error, while another may not check
+them (to maximize performance) and would just crash.
+
+Many ODP component areas provide an odp_xxx_capability() API that returns
+platform-specific information regarding supported features in that component.
+For best portability applications should always use these capability APIs.
+
+ODP APIs are described using abstract data types of which definition are left up
```
Comment (JereLeppanen) [inline include/README:24]: Opaque types, not abstract types.

File: include/README [hunk 3]
```diff
@@ -66,14 +107,13 @@ disabled. They should implement at least a set of types and values documented
 in ODP API specification headers. They are permitted to provide any platform
 specific optimizations (i.e. they might provide types and/or values that map
 directly onto the hardware details, they might provide inline functions to
-speed up execution of the application, etc). These headers MAY use ODP default
+speed up execution of the application, etc.). These headers MAY use ODP default
 ABI headers if they do fit.
 
-Rest of platform-specific headers
----------------------------------
+### Rest of platform-specific headers
```
Comment (JereLeppanen) [inline include/README:113]: Probably should be "The rest of the platform-specific headers". But "Other platform-specific headers" or "Additional platform-specific headers" might be better.

**Comment by MatiasElo** [comment]
V2:
- Fixed all Jere's comments

**Review by psavol — APPROVED** [review]

File: include/README [hunk 4]
```diff
@@ -1,63 +1,104 @@
...
+ODP API, one implementation may report an error, while another may not check
+them (to maximize performance) and would just crash.
+
+Many ODP component areas provide an odp_xxx_capability() API that returns
+platform-specific information regarding supported features in that component.
+For best portability applications should always use these capability APIs.
+
+ODP APIs are described using opaque data types of which definition are left up
+to the ODP implementation. For example, ODP packets are referenced by abstract
+handles of type odp_packet_t, and packet-related APIs take arguments of this
```
[the remark this answers is not in the document]
Comment (MatiasElo) [reply include/README]: Changed in V3.

**Review by JereLeppanen — APPROVED** [review]

**Pull request merged**